flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinator.java

```diff
@@ -107,7 +107,14 @@ public void start() throws Exception {
 		// The start sequence is the first task in the coordinator executor.
 		// We rely on the single-threaded coordinator executor to guarantee
 		// the other methods are invoked after the enumerator has started.
-		coordinatorExecutor.execute(() -> enumerator.start());
+		coordinatorExecutor.execute(() -> {
+			try {
+				enumerator.start();
+			} catch (Exception e) {
```

Comment:
Can we change this to `Throwable`? Then we also catch issues with dependencies, classpaths. It is what all other parts of Flink do and it is important. Otherwise one job with wrong dependencies kills a shared cluster.